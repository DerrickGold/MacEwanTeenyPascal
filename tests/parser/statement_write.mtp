begin
	(*write statement*)
	write(1+2*3);

	(*write statement, multiple expressions*)
	write(1+2*3, four, five, six);
end
.