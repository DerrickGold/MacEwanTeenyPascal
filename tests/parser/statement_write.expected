[Head] (Block)
    [Child] (Statement List)
        [Child] (Write Statement)
            [Child] 3, TOK_PLUS, +, (Expression, Simple Expression, Binary Adding Operator)
                [Child] 3, TOK_NUM, 1, (Simple Expression, Factor, Constant)
                [Child] 3, TOK_STAR, *, (Multiply Operator)
                    [Child] 3, TOK_NUM, 2, (Factor, Constant, Term)
                    [Child] 3, TOK_NUM, 3, (Factor, Constant, Term)
        [Sibling] (Write Statement)
            [Child] 6, TOK_PLUS, +, (Expression, Simple Expression, Binary Adding Operator)
                [Child] 6, TOK_NUM, 1, (Simple Expression, Factor, Constant)
                [Child] 6, TOK_STAR, *, (Multiply Operator)
                    [Child] 6, TOK_NUM, 2, (Factor, Constant, Term)
                    [Child] 6, TOK_NUM, 3, (Factor, Constant, Term)
            [Sibling] 6, TOK_ID, four, (Expression, Simple Expression, Factor, Constant, Simple Name)
            [Sibling] 6, TOK_ID, five, (Expression, Simple Expression, Factor, Constant, Simple Name)
            [Sibling] 6, TOK_ID, six, (Expression, Simple Expression, Factor, Constant, Simple Name)
        [Sibling] (Null Statement)
